client/team_project_assignment.go

```diff
@@ -16,14 +16,14 @@ const (
 type TeamProjectAssignmentPayload struct {
 	TeamId      string `json:"teamId"`
 	ProjectId   string `json:"projectId"`
-	ProjectRole Role   `json:"projectRole"`
+	ProjectRole Role   `json:"projectRole" tfschema:"role"`
```

Comment:
in the schema it's called "role".
tfschema resolves this for serialization and deserialization.